### PR TITLE
fix: handle promise result

### LINKED ISSUE #9699: Restore the no-floating-promise ESLint rule
### Is your enhancement related to a problem? Please describe

See https://github.com/containers/podman-desktop/blob/2db0de05182038207f103700ed74326d83792941/eslint.config.mjs#L305

### Describe the solution you'd like

Update code once rule is enabled

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #9699

### What does this PR do?

Update code to handle new no floating promises rule

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#9699

### How to test this PR?

Run unit tests

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by axel7083 — COMMENTED** [review]
Amazing job, some tests seems to be failing (expected with the number of change!)

**Comment by dgolovin** [comment]
I fixed some of it. Strange errors in learning-center which I implemented a while ago. Will take a look at it later.

**Comment by dgolovin** [comment]
rebased on top of origin/main.

**Comment by dgolovin** [comment]
One test remains in question. It seems that manual testing reveals the same problem with restarting the terminal for container and it should be investigated as follow up.

**Review by dgolovin — APPROVED** [review]
Follow up issue is needed for terminal restart issue.

**Comment by benoitf** [comment]
I would say the test needs to be fixed rather than being skipped

**Comment by dgolovin** [comment]
> I would say the test needs to be fixed rather than being skipped

I agree, but test failure probably related to #9771. My hope was to fix it quick to avoid conflict with #9288 fix.

File: packages/renderer/src/lib/pod/PodEmptyScreen.svelte [hunk 1]
```diff
@@ -10,8 +10,9 @@ const helloImage = 'quay.io/podman/hello:latest';
 const commandLine = `podman run -dt --pod new:${myFirstPod} ${helloImage}`;
 
 let inProgress = false;
+const provideInfosCopy = $providerInfos;
 
-$: selectedProviderConnection = $providerInfos
+$: selectedProviderConnection = provideInfosCopy
```
Comment (benoitf) [inline packages/renderer/src/lib/pod/PodEmptyScreen.svelte]: here I'm not sure to follow why there is this change regarding promise ?

File: packages/renderer/src/lib/pvc/PVCDetails.svelte [hunk 2]
```diff
@@ -33,12 +33,8 @@ onMount(() => {
...
     const matchingPVC = pvcs.find(pvc => pvc.metadata?.name === name && pvc.metadata?.namespace === namespace);
     if (matchingPVC) {
-      try {
-        pvc = pvcUtils.getPVCUI(matchingPVC);
-        loadDetails();
-      } catch (err) {
-        console.error(err);
-      }
+      pvc = pvcUtils.getPVCUI(matchingPVC);
+      loadDetails().catch((err: unknown) => console.error(`Error getting PVC details ${pvc.name}`, err));
```
Comment (benoitf) [inline packages/renderer/src/lib/pvc/PVCDetails.svelte:37]: could we have used here `async( pvcs)` and add the await before loadDetails()

File: packages/renderer/src/lib/window-control-buttons/ControlButtons.spec.ts [hunk 3]
```diff
@@ -46,7 +46,7 @@ describe.each([{ platform: 'linux' }, { platform: 'win32' }])('Platform($platfor
   test('Expect minimize is called', async () => {
     render(ControlButtons, { platform });
 
-    const minimizeMock = vi.fn();
+    const minimizeMock = vi.fn().mockResolvedValue(undefined);
```
Comment (benoitf) [inline packages/renderer/src/lib/window-control-buttons/ControlButtons.spec.ts]: not sure to follow why we have this change ?

File: packages/renderer/src/lib/window-control-buttons/ControlButtons.svelte [hunk 4]
```diff
@@ -4,15 +4,16 @@ import WindowControlButton from './ControlButton.svelte';
...
 }
 
 function maximize(): void {
-  window.windowMaximize();
+  window.windowMaximize().catch((err: unknown) => console.error('Error maximizing window', err));
 }
 
 function close(): void {
-  window.windowClose();
+  window.windowClose().catch((err: unknown) => console.error('Error closing window', err));
```
Comment (benoitf) [inline packages/renderer/src/lib/window-control-buttons/ControlButtons.svelte]: functions can't be async ?

File: packages/renderer/src/lib/window-control-buttons/ControlButtons.spec.ts (hunk 3 above)
Comment (benoitf) [reply packages/renderer/src/lib/window-control-buttons/ControlButtons.spec.ts]: usually our functions are async and then error will be thrown to the svelte button

so probably better to add async tot the method + await rather catch

File: packages/renderer/src/stores/tasks.ts [hunk 5]
```diff
@@ -30,7 +30,7 @@ export const tasksInfo: Writable<TaskInfo[]> = writable([]);
 
 // remove element from the store
 export function removeTask(taskId: string): void {
-  window.clearTask(taskId);
+  window.clearTask(taskId).catch((err: unknown) => console.error(`Error removing task ${taskId}`, err));
```
Comment (benoitf) [inline packages/renderer/src/stores/tasks.ts]: prefer async on the function + await rather than promise.catch

**Review by benoitf — APPROVED** [review]

File: packages/renderer/src/lib/pod/PodEmptyScreen.svelte (hunk 1 above)
Comment (jeffmaury) [reply packages/renderer/src/lib/pod/PodEmptyScreen.svelte]: Seems unrelated reverted by 7b3e7853b0dfb5515bfbe50164695c227632a782

File: packages/renderer/src/lib/pvc/PVCDetails.svelte (hunk 2 above)
Comment (jeffmaury) [reply packages/renderer/src/lib/pvc/PVCDetails.svelte:37]: Tried it but it it is not supported by the subscribe method signature

File: packages/renderer/src/lib/window-control-buttons/ControlButtons.svelte (hunk 4 above)
Comment (jeffmaury) [reply packages/renderer/src/lib/window-control-buttons/ControlButtons.svelte]: Done

**Pull request merged**